From bd72d4732aab994b1da7f8c4619802a17a519206 Mon Sep 17 00:00:00 2001
From: Brian Matherly <brian.matherly@yahoo.com>
Date: Sat, 9 Dec 2023 11:50:55 -0600
Subject: [PATCH] Add meta.media.aspect_ratio

Unlike the current "aspect_ratio" parameter, this one:
* accounts for rotation
* accounts for force_aspect_ratio
* propogates to XML as metadata

Needed to fix this issue:
https://forum.shotcut.org/t/slideshow-generator-cant-do-padding-of-videos/41889
---
 src/modules/avformat/producer_avformat.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/modules/avformat/producer_avformat.c b/src/modules/avformat/producer_avformat.c
index 7a0be161a..09007d93f 100644
--- a/src/modules/avformat/producer_avformat.c
+++ b/src/modules/avformat/producer_avformat.c
@@ -2973,6 +2973,7 @@ static void producer_set_up_video(producer_avformat self, mlt_frame frame)
                 mlt_properties_set_int(properties, "height", self->video_codec->height);
             }
             mlt_properties_set_double(frame_properties, "aspect_ratio", aspect_ratio);
+            mlt_properties_set_double(properties, "meta.media.aspect_ratio", aspect_ratio);
         }
         mlt_properties_set_int(frame_properties, "colorspace", self->yuv_colorspace);
         mlt_properties_set_int(frame_properties, "color_trc", self->color_trc);
@@ -3785,6 +3786,7 @@ static int producer_get_frame(mlt_producer producer, mlt_frame_ptr frame, int in
         // If video index is valid, get_image() must be called before the probe is complete
         mlt_properties_clear(MLT_PRODUCER_PROPERTIES(producer), "meta.media.width");
         mlt_properties_clear(MLT_PRODUCER_PROPERTIES(producer), "meta.media.height");
+        mlt_properties_clear(MLT_PRODUCER_PROPERTIES(producer), "meta.media.aspect_ratio");
         mlt_properties_clear(MLT_PRODUCER_PROPERTIES(producer), "meta.media.color_range");
         mlt_properties_clear(MLT_PRODUCER_PROPERTIES(producer), "meta.media.aspect_ratio");
         mlt_properties_clear(MLT_PRODUCER_PROPERTIES(producer), "meta.media.progressive");
